pkg/jobs/jobclient.go

```diff
@@ -0,0 +1,347 @@
+	if err != nil {
+		return &kubtest.ExecutionResult{
+			Status:       kubtest.ExecutionStatusError,
+			ErrorMessage: err.Error(),
+		}, err
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != v1.PodRunning {
+			if pod.Labels["job-name"] == jobName {
+				if err := wait.PollImmediate(time.Second, time.Duration(0)*time.Second, isPodRunning(c.ClientSet, pod.Name, c.Namespace)); err != nil {
+					return &kubtest.ExecutionResult{
+						Status:       kubtest.ExecutionStatusError,
+						ErrorMessage: err.Error(),
+					}, err
+				}
+			}
+			result, err = c.GetPodLogs(pod.Name, jobName, jobName)
+			if err != nil {
+				return &kubtest.ExecutionResult{
+					Status:       kubtest.ExecutionStatusError,
+					ErrorMessage: err.Error(),
+				}, err
+			}
+		}
+	}
+
+	return &kubtest.ExecutionResult{
+		Status:    kubtest.ExecutionStatusSuceess,
+		RawOutput: result,
+	}, nil
+}
+
+// connectToK8s returns ClientSet
+func connectToK8s() (*kubernetes.Clientset, error) {
+	var err error
+	var config *rest.Config
+	if cfg, exists := os.LookupEnv("KUBECONFIG"); !exists {
+		config, err = rest.InClusterConfig()
+	} else {
+		config, err = clientcmd.BuildConfigFromFlags("", cfg)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+// isPodRunning check if the pod in question is running state
+func isPodRunning(c *kubernetes.Clientset, podName, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		pod, err := c.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		switch pod.Status.Phase {
+		case v1.PodRunning, v1.PodSucceeded:
+			return true, nil
+		case v1.PodFailed:
+			return false, nil
+		}
+		return false, nil
+	}
+}
+
+func isPersistentVolumeBound(c *kubernetes.Clientset, podName, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		pv, err := c.CoreV1().PersistentVolumes().Get(context.Background(), podName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		switch pv.Status.Phase {
+		case v1.VolumeBound, v1.VolumeAvailable:
+			return true, nil
+		case v1.VolumeFailed:
+			return false, nil
+		}
+		return false, nil
+	}
+}
+
+func isPersistentVolumeClaimBound(c *kubernetes.Clientset, podName, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		pv, err := c.CoreV1().PersistentVolumeClaims(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		switch pv.Status.Phase {
+		case v1.ClaimBound:
+			return true, nil
+		case v1.ClaimLost:
+			return false, nil
+		}
+		return false, nil
+	}
+}
+
+func (c *JobClient) GetPodLogs(podName string, containerName string, endMessage string) (string, error) {
```

Comment:
We can structure the code to have the public methods first and then the private ones.

Reply:
This method will be used for watch so we need it to be public